gnd/src/main/java/com/google/android/gnd/repository/DataRepository.java

```diff
@@ -72,22 +74,65 @@ public DataRepository(
     this.remoteDataStore = remoteDataStore;
     this.dataSyncWorkManager = dataSyncWorkManager;
     this.cache = cache;
-    this.activeProject = BehaviorProcessor.create();
     this.uuidGenerator = uuidGenerator;
     this.localValueStore = localValueStore;
 
+    // BehaviorProcessor re-emits last requested project id to late subscribers.
+    this.activateProjectRequests = BehaviorProcessor.create();
+
+    // Load project when requested id changes, caching the last loaded project.
+    this.activeProjectStream =
+        activateProjectRequests
+            .distinctUntilChanged()
+            .doOnNext(id -> Log.v(TAG, "Requested project id changed: " + id))
+            .switchMap(this::onActivateProjectRequest)
+            .replay(1)
+            .refCount();
     streamFeaturesToLocalDb(remoteDataStore);
   }
 
+  private Flowable<Loadable<Project>> onActivateProjectRequest(Optional<String> projectId) {
+    // Empty id indicates intent to deactivate the current project. Used. on sign out.
+    if (projectId.isEmpty()) {
+      return Flowable.just(Loadable.notLoaded());
+    }
+    String id = projectId.get();
+
+    return syncProjectWithRemote(id)
+        .doOnSubscribe(__ -> Log.i(TAG, "Activating project " + id))
+        .onErrorResumeNext(
+            localDataStore
+                .getProjectById(id)
+                .doOnSubscribe(__ -> Log.i(TAG, "Falling back to local db"))
+                .toSingle())
```

Comment:
This would consume all kinds of exceptions even when it's not `FirebaseFirestoreException`. Do we want that?

Reply:
Good catch.. is that the only exception that would be meaningful here? In either case, we would want to abstract this type of failure (`[Local|Remote]DataStoreException`?), since the specific exceptions thrown are a detail of the local and remote data store implementations.

Reply:
I don't know what other kinds of exceptions are thrown. This was based on my observation while targeting only internet availability as the culprit for the error.